### PR TITLE
Fix the type annotations in the RTDB Query.on/off/once API.

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This fixes the nullability annotations in the RTDB Query.on/off API. The callback can be optional, but not the DataSnapshot. This is a breaking typing cleanup (as some types used to take "null"), but this updates the TS types to match the JS reference docs.

Fixes: https://github.com/firebase/firebase-js-sdk/issues/1188

### CODE REVIEW

File: packages/database-types/index.d.ts [hunk 1]
```diff
@@ -68,20 +68,20 @@ export interface Query {
   limitToLast(limit: number): Query;
   off(
     eventType?: EventType,
-    callback?: (a: DataSnapshot, b?: string | null) => any,
-    context?: Object | null
-  ): any;
+    callback?: (a: DataSnapshot, b?: string) => any,
```
Comment (mikelehen) [inline packages/database-types/index.d.ts]: `b` (aka `prevChildName`) will be null in the case of child_added events where there's no predecessor.

File: packages/database-types/index.d.ts [hunk 2]
```diff
@@ -68,20 +68,20 @@ export interface Query {
...
+    context?: Object
+  ): void;
   on(
     eventType: EventType,
-    callback: (a: DataSnapshot | null, b?: string) => any,
-    cancelCallbackOrContext?: Object | null,
-    context?: Object | null
-  ): (a: DataSnapshot | null, b?: string) => any;
+    callback: (a: DataSnapshot, b?: string) => any,
```
Comment (mikelehen) [inline packages/database-types/index.d.ts]: `b` (aka `prevChildName`) will be null in the case of child_added events where there's no predecessor.

File: packages/database/src/core/view/Event.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ export class DataEvent implements Event {
       | ' child_removed',
     public eventRegistration: EventRegistration,
     public snapshot: DataSnapshot,
-    public prevName?: string | null
+    public prevName?: string
```
Comment (mikelehen) [inline packages/database/src/core/view/Event.ts]: can be null

**Comment by mikelehen** [comment]
FYI- Still LGTM after changelog addition.

**Comment by schmidt-sebastian** [comment]
This will be merged into the Firebase I/O release as it contains minor breaking changes.

**Comment by schmidt-sebastian** [comment]
@Feiyang1 Can you merge this with the version bump?

**Pull request merged**